### PR TITLE
Add parsing and objdump support for updated "linking" section

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

**Comment by sbc100** [comment]
Should be ready to go now..

**Comment by sbc100** [comment]
Added a test.

File: test/binary/linking_section.txt [hunk 1]
```diff
@@ -0,0 +1,26 @@
+;;; TOOL: run-gen-wasm
+;;; FLAGS: --objdump
+magic
+version
+section("linking") {
+  subsection[1]
+  length[1]
+  stack_galbal[3]
```
Comment (binji) [inline test/binary/linking_section.txt]: global?

File: test/run-gen-wasm.py [hunk 2]
```diff
@@ -46,30 +46,38 @@ def main(args):
...
+      error_cmdline=options.error_cmdline)
+
   wasm2wast = utils.Executable(
       find_exe.GetWasm2WastExecutable(options.bindir),
       error_cmdline=options.error_cmdline)
   wasm2wast.AppendOptionalArgs({
       '--no-debug-names': options.no_debug_names,
-      '--generate-names': options.generate_names,
   })
+  wasm2wast.AppendOptionalArgs({'--verbose': options.verbose,})
```
Comment (binji) [inline test/run-gen-wasm.py]: Why not include this in the previous one?

File: test/run-gen-wasm.py [hunk 3]
```diff
@@ -46,30 +46,38 @@ def main(args):
...
+  objdump = utils.Executable(
+      find_exe.GetWasmdumpExecutable(options.bindir),
+      error_cmdline=options.error_cmdline)
+
   wasm2wast = utils.Executable(
       find_exe.GetWasm2WastExecutable(options.bindir),
       error_cmdline=options.error_cmdline)
   wasm2wast.AppendOptionalArgs({
       '--no-debug-names': options.no_debug_names,
-      '--generate-names': options.generate_names,
```
Comment (binji) [inline test/run-gen-wasm.py]: Isn't this used?

File: test/run-gen-wasm.py (hunk 2 above)
Comment (binji) [inline test/run-gen-wasm.py]: Oh, is this meant to be `objdump.AppendOptionalArgs`?

File: test/run-gen-wasm.py (hunk 3 above)
Comment (sbc100) [inline test/run-gen-wasm.py]: Yes. Not used.

File: test/run-gen-wasm.py (hunk 2 above)
Comment (sbc100) [inline test/run-gen-wasm.py]: No, I just moved this line up.  Not that --verbose is not the same as .verbose attribute below.

Comment (binji) [inline test/run-gen-wasm.py]: Oh, I see.

**Comment by binji** [comment]
lgtm

**Pull request merged**